src/System.Device.Gpio/System/Device/Gpio/Drivers/LibgpiodCapiDriver.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Runtime.InteropServices;
+
+namespace System.Device.Gpio.Drivers
+{
+    public class LibgpiodCapiDriver : GpioDriver
+    {
+        private const string BasePath = "/dev/";
+
+        public string GpioChip { get; set; } = "gpiochip0";
+
+        private SafeChipHandle _chip;
```

Comment:
I don't think we can go with this current approach where each driver represents one chip only. I believe that instead we should have the driver represent all of the chips, and abrstract that concept from the user. User only knows about pin numbers, and will only tell you they want to open pin 18, they don't know which chip this pin is on and it shouldn't be up to them to figure that out. Instead, that logic should be traversed by the driver.

Reply:
Do you know what this chips represent in the device? By my understanding/assumption each chip represents each programmable ports available in the device. For example gpiochip0 is the parallel port on the rasp pie, with command gpioinfo i see 2 more chips except that which i assume representing other ports like USB or LAN or HDMI, if that is true we shouldn't logically integrate them, but as i said its just my assumption, need to dig more, if anyone knows aboutn this please let me know